Automattic/wp-calypso
# Update Jetpack Cloud Backup Time Logic

#### Changes proposed in this Pull Request

* Switch `DatePicker` from semi-controlled to controlled component
* Use Moment for `getBackupAttemptsForDate`  
* Add a few tests for `getBackupAttemptsForDate`
* Change `date` state in `BackupPage` to Date

#### Testing instructions



1. run new tests via `npm run test-client -- --testPathPattern jetpack-cloud/sections/backup` and confirm they pass
2. Load up this PR and make sure the component appears for each day that can be selected, and that there are no console errors related to this component
3. Ensure that shuttling back and forth with the date picker changes the date for the backup status component, and that dates with missing backups explain what went wrong in the error message.
